libs/core/tests/integration/state_machine_tests.cpp

```diff
@@ -0,0 +1,177 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2020 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <utility>
+
+#include "core/reactor.hpp"
+#include "core/state_machine.hpp"
+
+#include "gmock/gmock.h"
+
+using namespace fetch;
+using namespace testing;
+
+class ReactorTests : public Test
+{
+public:
+  enum class State : uint8_t
+  {
+    A,
+    B,
+    C,
```

Comment:
IIRC trailing comma is non-standard in C++.

Reply:
fixed